bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java

```diff
@@ -99,8 +102,9 @@
      * @param localIP local IP address from networkAddressService
      * @param httpPort from httpService
      */
-    public ShellyBaseHandler(Thing thing, ShellyBindingConfiguration bindingConfig,
-            @Nullable ShellyCoapServer coapServer, String localIP, int httpPort) {
+    public ShellyBaseHandler(Thing thing, ShellyListenerManager listenerManager,
```

Comment:
Why do you pass the listenerManager to the handler when it isn't at all used in the constructor? Can't this parameter be removed? It would be much nicer, if the listenerManager was only used within the handler factory - and in this case, you could also get rid off the ShellyListenerManager class again, which honestly looks a bit awkward.

Reply:
the factory registers the new thing handler to the onEvent listener list (as requested)
how should the thing handler deregister from that list on dispose()?

Reply:
Why does the factory do the registration, while the handler does the unregistration? Does not look very symmetric. If the factory registers it, it can also unregister it (in the `removeHandler` method) and there would be no need to pass the listener list to the handler, which anyhow looks awkward, as mentioned above.

Reply:
solved, implemented removeHandler()